modules/appeals_api/spec/requests/v1/appeals_controller_spec.rb

```diff
@@ -15,97 +15,121 @@
+          it 'returns a 422 error' do
+            expect(response).to have_http_status(:unprocessable_entity)
+          end
         end
-      end
 
-      describe 'when veteran SSN is not found by ICN in MPI' do
-        let(:mpi_cassette_name) { 'mpi/find_candidate/icn_not_found' }
+        describe 'when veteran SSN is not found by ICN in MPI' do
+          let(:mpi_cassette_name) { 'mpi/find_candidate/icn_not_found' }
 
-        it 'returns a 404 error with a message that does not reference SSN' do
-          expect(response).to have_http_status(:not_found)
-          expect(error['detail']).not_to include('SSN')
+          it 'returns a 404 error with a message that does not reference SSN' do
+            expect(response).to have_http_status(:not_found)
+            expect(error['detail']).not_to include('SSN')
+          end
         end
-      end
 
-      describe 'when MPI throws an error' do
-        let(:mpi_cassette_name) { 'mpi/find_candidate/internal_server_error' }
+        describe 'when MPI throws an error' do
+          let(:mpi_cassette_name) { 'mpi/find_candidate/internal_server_error' }
 
-        it 'returns a 502 error instead' do
-          expect(response).to have_http_status(:bad_gateway)
+          it 'returns a 502 error instead' do
+            expect(response).to have_http_status(:bad_gateway)
+          end
         end
-      end
 
-      describe 'when veteran is not found by SSN in caseflow' do
-        let(:caseflow_cassette_name) { 'caseflow/not_found' }
+        describe 'when veteran is not found by SSN in caseflow' do
+          let(:caseflow_cassette_name) { 'caseflow/not_found' }
 
-        it 'returns a 404 error with a message that does not reference SSN' do
-          expect(response).to have_http_status(:not_found)
-          expect(error['detail']).not_to include('SSN')
+          it 'returns a 404 error with a message that does not reference SSN' do
+            expect(response).to have_http_status(:not_found)
+            expect(error['detail']).not_to include('SSN')
+          end
         end
-      end
 
-      describe 'when caseflow throws a 500 error' do
-        let(:caseflow_cassette_name) { 'caseflow/server_error' }
+        describe 'when caseflow throws a 500 error' do
+          let(:caseflow_cassette_name) { 'caseflow/server_error' }
+
+          it 'returns a 502 error instead' do
+            expect(response).to have_http_status(:bad_gateway)
+          end
+        end
+      end
 
-        it 'returns a 502 error instead' do
-          expect(response).to have_http_status(:bad_gateway)
+      it_behaves_like('an endpoint with OpenID auth', scopes: described_class::OAUTH_SCOPES[:GET]) do
+        def make_request(auth_header)
+          VCR.use_cassette(caseflow_cassette_name) do
+            VCR.use_cassette(mpi_cassette_name) do
+              get(path, params: { icn: }, headers: auth_header.merge(headers))
+            end
+          end
```

Comment:
These were the lines added to this spec file. (The rest were just indents under a new `describe '#index'` block.)